app/models/team.rb

```diff
@@ -5,15 +5,30 @@ class Team < ApplicationRecord
   has_one :team_node, foreign_key: :node_object_id, dependent: :destroy
   has_many :signed_up_teams, dependent: :destroy
   has_many :bids, dependent: :destroy
+  has_many :participants, through: :teams_users
   has_paper_trail
 
-  scope :find_team_for_assignment_and_user, lambda { |assignment_id, user_id|
-    joins(:teams_users).where('teams.parent_id = ? AND teams_users.user_id = ?', assignment_id, user_id)
-  }
+   def self.find_team_for_assignment_and_user(assignment_id, user_id)
+    participant = Participant.find_by(user_id: user_id, parent_id: assignment_id)
+    team_user = TeamsUser.find_by(participant_id: participant.id)
+    team_user = TeamsUser.where(user_id: user_id).find { |team_user_obj| team_user_obj.team.parent_id == assignment_id } if team_user.nil?
+    Team.find(team_user.team_id)
+  end
 
+ # E2283 Remove this function when user_id is removed from teams_users table
+  # teams_participants table now stores participant_id instead of user_id but, older entries
+  # have user_id mapping. This function loads participants details directly from the mapping
+  # as well as from user_id if some entries use user_id mapping.
   # Get the participants of the given team
   def participants
-    users.where(parent_id: parent_id || current_user_id).flat_map(&:participants)
+    # load directly using participant_ids
+    participants_list_from_ids = Participant.where(id: participant_ids)
+
+    unless participants_list_from_ids.nil?
+      participants_list = []
```

Comment:
Can you use [append method](https://www.geeksforgeeks.org/ruby-array-append-function/)

Reply:
`append` is not working. Used `push` to append the array